Sources/DealiDesignKit/Components/Alert/DealiAlert.swift

```diff
@@ -0,0 +1,500 @@
+            self.cancelButton.then {
+                $0.style = .medium(style: .outlined)
+                $0.title = cancelButtonTitle
+                $0.addTarget(self, action: #selector(cancelButtonAction), for: .touchUpInside)
+            }.snp.makeConstraints {
+                $0.top.bottom.equalToSuperview()
+            }
+        }
+        
+        if let confirmButtonTitle = self.confirmButtonTitle {
+            buttonStackView.addArrangedSubview(self.confirmButton)
+            self.confirmButton.then {
+                $0.style = .medium(style: .filled)
+                $0.title = confirmButtonTitle
+                $0.addTarget(self, action: #selector(confirmButtonAction), for: .touchUpInside)
+            }.snp.makeConstraints {
+                $0.top.bottom.equalToSuperview()
+            }
+        }
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        // 스크롤뷰 사이즈 조정
+        if self.messageContentStackView.subviews.count > 0 {
+            self.messageContentScrollView.layoutIfNeeded()
+            var contentHeight: CGFloat = self.messageContentScrollView.contentSize.height
+            if contentHeight > 306 {
+                contentHeight = 306
+            }
+            self.messageContentScrollView.snp.updateConstraints {
+                $0.height.equalTo(contentHeight)
+            }
+        }
+        
+    }
+    
+    @objc func cancelButtonAction() {
+        self.dismiss(animated: true) { [weak self] in
+            if let self = self, let action = self.cancelAction {
+                action()
+            }
+        }
+    }
+    
+    @objc func confirmButtonAction() {
+        self.dismiss(animated: true) { [weak self] in
+            if let self = self, let action = self.confirmAction {
+                action()
+            }
+        }
+    }
+    
+    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesEnded(touches, with: event)
+        guard let touch = touches.first, self.contentView.bounds.contains(touch.location(in: self.contentView)) == false  else { return }
+        if self.closeAlertOnOutsideTouch == true {
+            self.dismiss(animated: true, completion: nil)
+        } else {
+            if self.cancelAndCloseOnOutsideTouch == true {
+                self.cancelButtonAction()
+            }
+        }
+    }
+}
+
+extension UIViewController {
+    private struct UIViewControllerPreview: UIViewControllerRepresentable {
+        let viewController: UIViewController
+
+                func makeUIViewController(context: Context) -> UIViewController {
+                    return viewController
+                }
+
+                func updateUIViewController(_ uiViewController: UIViewController, context: Context) {
+                }
+    }
+    
+    func showPreview() -> some View {
+        UIViewControllerPreview(viewController: self)
+        }
+}
+
```

Comment:
인덴트 조정이랑 SwiftUI 때문에 디버그플래그 걸어주세요~ 
그리고 이 부분은 모든 뷰컨트롤러에서 사용할 수 있는 부분이니까 Util 폴더로 옮기는 것이 낫지 않을까요? UIView 에 대한 비슷한 기능도 거기 모여있어서요

```suggestion
#if canImport(SwiftUI) && DEBUG
extension UIViewController {
    private struct UIViewControllerPreview: UIViewControllerRepresentable {
        let viewController: UIViewController
        
        func makeUIViewController(context: Context) -> UIViewController {
            return viewController
        }
        
        func updateUIViewController(_ uiViewController: UIViewController, context: Context) {
        }
    }
    
    func showPreview() -> some View {
        UIViewControllerPreview(viewController: self)
    }
}
#endif
```

Reply:
수정했음